lib/internal/util/inspect_colors.js

```diff
@@ -0,0 +1,505 @@
+      for (let i = 0; i < lineTypes.length; ++i) {
+        const lineType = lineTypes[i];
+        if (
+          ArrayPrototypeIncludes(
+            ['line-through', 'overline', 'underline'],
+            lineType,
+          )
+        ) {
+          ArrayPrototypePush(css.textDecorationLine, lineType);
+        }
+      }
+    } else if (key === 'text-decoration-color') {
+      const color = parseCssColor(value);
+      if (color != null) {
+        css.textDecorationColor = color;
+      }
+    } else if (key === 'text-decoration') {
+      css.textDecorationColor = null;
+      css.textDecorationLine = [];
+      const args = StringPrototypeSplit(value, SPACE_PATTERN);
+      for (let i = 0; i < args.length; ++i) {
+        const arg = args[i];
+        const maybeColor = parseCssColor(arg);
+        if (maybeColor != null) {
+          css.textDecorationColor = maybeColor;
+        } else if (
+          ArrayPrototypeIncludes(['line-through', 'overline', 'underline'], arg)
+        ) {
+          ArrayPrototypePush(css.textDecorationLine, arg);
+        }
+      }
+    }
+  }
+
+  return css;
+}
+
+// Copyright 2018-2023 the Deno authors. All rights reserved. MIT license.
+// https://github.com/denoland/deno/blob/ece2a3de5b19588160634452638aa656218853c5/ext/console/01_console.js#L2928
+function colorEquals(color1, color2) {
+  return color1?.[0] === color2?.[0] && color1?.[1] === color2?.[1] &&
+    color1?.[2] === color2?.[2];
+}
+
+// Copyright 2018-2023 the Deno authors. All rights reserved. MIT license.
+// https://github.com/denoland/deno/blob/ece2a3de5b19588160634452638aa656218853c5/ext/console/01_console.js#L2933
+function cssToAnsi(css, prevCss = null) {
+  prevCss = prevCss ?? getDefaultCss();
+  let ansi = '';
+  if (!colorEquals(css.backgroundColor, prevCss.backgroundColor)) {
+    // Why not use util.inspect.colors? Because it's user mutable. The colors
+    // here would be changeable like
+    // `util.inspect.colors.black = util.inspect.colors.red` but what about
+    // the colorKeywords map like "color:salmon"? Using util.inspect.colors
+    // would introduce a weird edge-case and require more cognitive overhead
+    // to read & understand since it's in a different module.
+    if (css.backgroundColor == null) {
+      ansi += '\x1b[49m';
+    } else if (css.backgroundColor === 'black') {
+      ansi += '\x1b[40m';
+    } else if (css.backgroundColor === 'red') {
+      ansi += '\x1b[41m';
+    } else if (css.backgroundColor === 'green') {
+      ansi += '\x1b[42m';
+    } else if (css.backgroundColor === 'yellow') {
+      ansi += '\x1b[43m';
+    } else if (css.backgroundColor === 'blue') {
+      ansi += '\x1b[44m';
+    } else if (css.backgroundColor === 'magenta') {
+      ansi += '\x1b[45m';
+    } else if (css.backgroundColor === 'cyan') {
+      ansi += '\x1b[46m';
+    } else if (css.backgroundColor === 'white') {
+      ansi += '\x1b[47m';
```

Comment:
nit: could these be moved into a table similar to the colors table at the top of the file?